### ISSUE TITLE
train.py: error: argument --manualseed: invalid int value: '{0..2}'

### ISSUE THREAD

**Issue by GuokaiLiu** [samet-akcay/ganomaly#12]
Thanks for your great work!
Not familiar with **sh** command,
When I run the command: `sh experiments/run_mnist.sh`,
the following comes out:
```
Manual Seed: {0..2}
Running mnist_{0..9}
usage: train.py [-h] [--dataset DATASET] [--dataroot DATAROOT]
                [--batchsize BATCHSIZE] [--workers WORKERS] [--droplast]
                [--isize ISIZE] [--nc NC] [--nz NZ] [--ngf NGF] [--ndf NDF]
                [--extralayers EXTRALAYERS] [--gpu_ids GPU_IDS] [--ngpu NGPU]
                [--name NAME] [--model MODEL]
                [--display_server DISPLAY_SERVER]
                [--display_port DISPLAY_PORT] [--display_id DISPLAY_ID]
                [--display] [--outf OUTF] [--manualseed MANUALSEED]
                [--anomaly_class ANOMALY_CLASS] [--proportion PROPORTION]
                [--metric METRIC] [--print_freq PRINT_FREQ]
                [--save_image_freq SAVE_IMAGE_FREQ] [--save_test_images]
                [--load_weights] [--resume RESUME] [--phase PHASE]
                [--iter ITER] [--niter NITER] [--beta1 BETA1] [--lr LR]
                [--w_bce W_BCE] [--w_rec W_REC] [--w_enc W_ENC]
train.py: error: argument --manualseed: invalid int value: '{0..2}'
```
What should I do?
Thank you very much!

**Comment by samet-akcay** [comment]
Sounds like bash cannot read the values as integers. Can you try the following from your terminal: `python train.py --dataset mnist --isize 32 --nc 1 --niter 15 --anomaly_class 0`, which trains mnist dataset when the anomalous class is 0.  If this works, you can make sure that the repo on your machine works fine.

**Comment by GuokaiLiu** [comment]
Thanks for your timely response :  )
It seems ok as follows:
```
python train.py --dataset mnist --isize 32 --nc 1 --niter 2 --anomaly_class 0
/home/brucelau/anaconda3/envs/fcn/lib/python3.5/site-packages/torchvision/transforms/transforms.py:188: UserWarning: The use of the transforms.Scale transform is deprecated, please use transforms.Resize instead.
  "please use transforms.Resize instead.")
>> Training model Ganomaly.
>> Training model Ganomaly. Epoch 1/2                                           
   Avg Run Time (ms/batch): 15.247 AUC: 0.869 max AUC: 0.869
>> Training model Ganomaly. Epoch 2/2                                           
   Avg Run Time (ms/batch): 14.507 AUC: 0.857 max AUC: 0.869
>> Training model Ganomaly.[Done]
```

I'll close this issue and dive into the code.
Thank you!

**Comment by ab2010do** [comment]
hi  samet-akcay 
i have got the problem when training mobilefacenet
by using this command : CUDA_VISIBLE_DEVICES='0,1,2,3' python -u train_softmax.py --network y1 --loss-type 4 --margin-m 0.5 --data-dir ../datasets/faces_ms1m_112x112 --prefix ../model-y1 --emb-size = 128
and i got this problem
usage: train_softmax.py [-h] [--data-dir DATA_DIR] [--prefix PREFIX]
                        [--pretrained PRETRAINED] [--ckpt CKPT]
                        [--loss-type LOSS_TYPE] [--verbose VERBOSE]
                        [--max-steps MAX_STEPS] [--end-epoch END_EPOCH]
                        [--network NETWORK] [--image-size IMAGE_SIZE]
                        [--version-se VERSION_SE]
                        [--version-input VERSION_INPUT]
                        [--version-output VERSION_OUTPUT]
                        [--version-unit VERSION_UNIT]
                        [--version-multiplier VERSION_MULTIPLIER]
                        [--version-act VERSION_ACT]
                        [--use-deformable USE_DEFORMABLE] [--lr LR]
                        [--lr-steps LR_STEPS] [--wd WD]
                        [--fc7-wd-mult FC7_WD_MULT]
                        [--fc7-lr-mult FC7_LR_MULT] [--fc7-no-bias]
                        [--bn-mom BN_MOM] [--mom MOM] [--emb-size EMB_SIZE]
                        [--per-batch-size PER_BATCH_SIZE]
                        [--margin-m MARGIN_M] [--margin-s MARGIN_S]
                        [--margin-a MARGIN_A] [--margin-b MARGIN_B]
                        [--easy-margin EASY_MARGIN] [--margin MARGIN]
                        [--beta BETA] [--beta-min BETA_MIN]
                        [--beta-freeze BETA_FREEZE] [--gamma GAMMA]
                        [--power POWER] [--scale SCALE]
                        [--rand-mirror RAND_MIRROR] [--cutoff CUTOFF]
                        [--color COLOR] [--images-filter IMAGES_FILTER]
                        [--target TARGET] [--ce-loss]
train_softmax.py: error: argument --emb-size: invalid int value: '='
how i can fix it?

**Comment by JiangAndy** [comment]
@GuokaiLiu 
I think it may be the version of python, you can try it using python3

**Comment by rafaelbidese** [comment]
If you are running on Ubuntu, the default shell is dash, so the  {0..2} increments don't work there. If you run the line as `bash experiments/run_cifar.sh` it will work smoothly.

I got that information from [stackoverflow](https://stackoverflow.com/questions/22863783/bash-script-cant-get-for-loop-working).
